File: Hwang/week3/boj_1072.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.on("line", function (line) {
    let [X, Y] = line.split(" ").map(Number);
    let Z = Math.floor((Y * 100) / X); // 현재 승률

    if (Z >= 99) {
        console.log(-1); // 99% 이상이면 더 이상 승률을 올릴 수 없음
        rl.close();
        return;
    }

    let left = 0, right = 1000000000, answer = -1;

    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        let newZ = Math.floor(((Y + mid) * 100) / (X + mid)); // 새로운 승률 계산

        if (newZ > Z) { 
            answer = mid;
            right = mid - 1; // 더 적은 경기 수를 찾기 위해 범위를 줄임
        } else {
            left = mid + 1; // 승률이 오르지 않으면 mid 값을 늘려야 함
        }
    }

    console.log(answer);
    rl.close();
});

File: Park/week3/boj_1072.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class boj_1072 {
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");

        long X = Long.parseLong(input[0]); // 총 게임 수
        long Y = Long.parseLong(input[1]); // 이긴 게임 수

        long Z = (Y * 100) / X; // 현재 승률

        // 승률이 변하지 않는다면 -1 출력
        if (Z >= 99) {
            System.out.println(-1);
            return;
        }
        
        // 추가 승히 횟수 최소 값 0 , 최대 x번 까지 이길 수 있다고 가정
        // 승률을 올리기 위해선 최대 x번 까지 이겨야 할 극단적인 경우가 있음. 따라서 x는 right
        long left = 0, right = X, answer = -1;

        while (left <= right) {
            long mid = (left + right) / 2;
            long newZ = ((Y + mid) * 100) / (X + mid);

            if (newZ > Z) { // 승률이 증가하면 mid를 줄여야 함
                answer = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        System.out.println(answer);
    }
}
